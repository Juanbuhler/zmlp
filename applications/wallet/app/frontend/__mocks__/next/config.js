import dotenv from 'dotenv'

dotenv.config()

const getConfig = () => ({
  serverRuntimeConfig: process.env,
  publicRuntimeConfig: process.env,
})

export default getConfig
