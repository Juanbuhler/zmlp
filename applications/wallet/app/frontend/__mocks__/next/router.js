/**
 * on
 */

let mockOnFunction = () => {}

export const __setMockOnFunction = fn => {
  mockOnFunction = fn
}

/**
 * off
 */

let mockOffFunction = () => {}

export const __setMockOffFunction = fn => {
  mockOffFunction = fn
}

/**
 * Router
 */

const Router = {
  events: {
    on: (...args) => {
      mockOnFunction(...args)
    },
    off: (...args) => {
      mockOffFunction(...args)
    },
  },
}

export default Router
