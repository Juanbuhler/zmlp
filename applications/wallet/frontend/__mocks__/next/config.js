const getConfig = () => ({
  serverRuntimeConfig: process.env,
  publicRuntimeConfig: process.env,
})

export default getConfig
