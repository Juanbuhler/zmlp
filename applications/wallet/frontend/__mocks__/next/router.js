/**
 * useRouter
 */

let mockUseRouter = {
  query: {
    action: '',
  },
}

export const __setUseRouter = data => {
  mockUseRouter = data
}

export const useRouter = () => mockUseRouter

/**
 * push
 */

let mockPushFunction = () => {}

export const __setMockPushFunction = fn => {
  mockPushFunction = fn
}

/**
 * on
 */

let mockOnFunction = () => {}

export const __setMockOnFunction = fn => {
  mockOnFunction = fn
}

/**
 * off
 */

let mockOffFunction = () => {}

export const __setMockOffFunction = fn => {
  mockOffFunction = fn
}

/**
 * Router
 */

const Router = {
  push: (...args) => {
    mockPushFunction(...args)
  },
  events: {
    on: (...args) => {
      mockOnFunction(...args)
    },
    off: (...args) => {
      mockOffFunction(...args)
    },
  },
}

export default Router
