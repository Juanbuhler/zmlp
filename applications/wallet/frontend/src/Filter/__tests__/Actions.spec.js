import TestRenderer, { act } from 'react-test-renderer'

import FilterActions from '../Actions'

const noop = () => () => {}

const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'
const ASSET_ID = ''

describe('<FilterActions />', () => {
  it('should render properly', () => {
    const filters = [{ attribute: 'clip.length', type: 'range' }]

    const mockRouterPush = jest.fn()

    require('next/router').__setMockPushFunction(mockRouterPush)

    const component = TestRenderer.create(
      <FilterActions
        pathname="/[projectId]/visualizer"
        projectId={PROJECT_ID}
        assetId={ASSET_ID}
        filters={filters}
        filter={filters[0]}
        filterIndex={0}
      />,
    )

    expect(component.toJSON()).toMatchSnapshot()

    // Disable filter
    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Disable Filter' })
        .props.onClick({ preventDefault: noop, stopPropagation: noop })
    })

    const encodedDisable = btoa(
      JSON.stringify([
        { attribute: 'clip.length', type: 'range', isDisabled: true },
      ]),
    )

    expect(component.toJSON()).toMatchSnapshot()
    expect(mockRouterPush).toHaveBeenCalledWith(
      {
        pathname: '/[projectId]/visualizer',
        query: {
          projectId: '76917058-b147-4556-987a-0a0f11e46d9b',
          id: '',
          query: encodedDisable,
        },
      },
      `/76917058-b147-4556-987a-0a0f11e46d9b/visualizer?query=${encodedDisable}`,
    )

    // Delete filter
    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Delete Filter' })
        .props.onClick({ preventDefault: noop, stopPropagation: noop })
    })

    expect(component.toJSON()).toMatchSnapshot()
    expect(mockRouterPush).toHaveBeenCalledWith(
      {
        pathname: '/[projectId]/visualizer',
        query: {
          projectId: '76917058-b147-4556-987a-0a0f11e46d9b',
          id: '',
          query: '',
        },
      },
      '/76917058-b147-4556-987a-0a0f11e46d9b/visualizer',
    )
  })

  it('should render disabled properly', () => {
    const filters = [
      { attribute: 'clip.length', type: 'range', isDisabled: true },
    ]

    const mockRouterPush = jest.fn()

    require('next/router').__setMockPushFunction(mockRouterPush)

    const component = TestRenderer.create(
      <FilterActions
        pathname="/[projectId]/visualizer"
        projectId={PROJECT_ID}
        assetId={ASSET_ID}
        filters={filters}
        filter={filters[0]}
        filterIndex={0}
      />,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
