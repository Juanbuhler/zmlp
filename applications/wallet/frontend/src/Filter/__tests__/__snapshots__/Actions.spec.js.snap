// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterActions /> should render disabled properly 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  opacity: 0;
}

.emotion-3 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-3:hover,
.emotion-3.focus-visible:focus {
  color: #ffffff;
}

.emotion-3:hover svg,
.emotion-3.focus-visible:focus svg {
  opacity: 1;
}

.emotion-3[aria-disabled=true] {
  color: #808080;
}

.emotion-3:hover,
.emotion-3.focus-visible:focus {
  background-color: #3b3b3b;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  color: #ffffff;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  opacity: 1;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  background-color: #3b3b3b;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  color: #d6680b;
}

.emotion-0 {
  opacity: 1;
}

<div
  className="emotion-4"
>
  <button
    aria-label="Enable Filter"
    className="emotion-1"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-0"
      color="#d6680b"
      data-file-name="hidden.svg"
      height={20}
    />
  </button>
  <button
    aria-label="Delete Filter"
    className="emotion-3"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-2"
      color="#808080"
      data-file-name="cross.svg"
      height={20}
    />
  </button>
</div>
`;

exports[`<FilterActions /> should render properly 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  color: #ffffff;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  opacity: 1;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  background-color: #3b3b3b;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  color: #ffffff;
}

.emotion-0 {
  opacity: 0;
}

.emotion-3 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-3:hover,
.emotion-3.focus-visible:focus {
  color: #ffffff;
}

.emotion-3:hover svg,
.emotion-3.focus-visible:focus svg {
  opacity: 1;
}

.emotion-3[aria-disabled=true] {
  color: #808080;
}

.emotion-3:hover,
.emotion-3.focus-visible:focus {
  background-color: #3b3b3b;
}

<div
  className="emotion-4"
>
  <button
    aria-label="Disable Filter"
    className="emotion-1"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-0"
      color="#808080"
      data-file-name="hidden.svg"
      height={20}
    />
  </button>
  <button
    aria-label="Delete Filter"
    className="emotion-3"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-0"
      color="#808080"
      data-file-name="cross.svg"
      height={20}
    />
  </button>
</div>
`;

exports[`<FilterActions /> should render properly 2`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  color: #ffffff;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  opacity: 1;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  background-color: #3b3b3b;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  color: #ffffff;
}

.emotion-0 {
  opacity: 0;
}

.emotion-3 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-3:hover,
.emotion-3.focus-visible:focus {
  color: #ffffff;
}

.emotion-3:hover svg,
.emotion-3.focus-visible:focus svg {
  opacity: 1;
}

.emotion-3[aria-disabled=true] {
  color: #808080;
}

.emotion-3:hover,
.emotion-3.focus-visible:focus {
  background-color: #3b3b3b;
}

<div
  className="emotion-4"
>
  <button
    aria-label="Disable Filter"
    className="emotion-1"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-0"
      color="#808080"
      data-file-name="hidden.svg"
      height={20}
    />
  </button>
  <button
    aria-label="Delete Filter"
    className="emotion-3"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-0"
      color="#808080"
      data-file-name="cross.svg"
      height={20}
    />
  </button>
</div>
`;

exports[`<FilterActions /> should render properly 3`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  color: #ffffff;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  opacity: 1;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  background-color: #3b3b3b;
}

.emotion-1:hover svg,
.emotion-1.focus-visible:focus svg {
  color: #ffffff;
}

.emotion-0 {
  opacity: 0;
}

.emotion-3 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  color: #808080;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-3:hover,
.emotion-3.focus-visible:focus {
  color: #ffffff;
}

.emotion-3:hover svg,
.emotion-3.focus-visible:focus svg {
  opacity: 1;
}

.emotion-3[aria-disabled=true] {
  color: #808080;
}

.emotion-3:hover,
.emotion-3.focus-visible:focus {
  background-color: #3b3b3b;
}

<div
  className="emotion-4"
>
  <button
    aria-label="Disable Filter"
    className="emotion-1"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-0"
      color="#808080"
      data-file-name="hidden.svg"
      height={20}
    />
  </button>
  <button
    aria-label="Delete Filter"
    className="emotion-3"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-0"
      color="#808080"
      data-file-name="cross.svg"
      height={20}
    />
  </button>
</div>
`;
